pkg/apis/elasticsearch/v1alpha1/types.go

```diff
@@ -80,8 +80,10 @@ type ElasticsearchNodeStatus struct {
 }
 
 type ElasticsearchNodeUpgradeStatus struct {
-	UnderUpgrade UpgradeStatus             `json:"underUpgrade,omitempty"`
-	UpgradePhase ElasticsearchUpgradePhase `json:"upgradePhase,omitempty"`
+	ScheduledForUpgrade  ConditionStatus           `json:"scheduledUpgrade,omitempty"`
```

Comment:
condensed how? they all denote different states

Reply:
Typically, something is in a singular state represented by what that state is like "ScheduledForUpgrade" and it transitions to some other state.  I question our need to represent states as a boolean because, for example, it results in field expansion everytime we add  a new state; we should become better at defining our API since its log lived.  In this case, cant we define state transition:  shedulededforupgrade -> underupgrade->scheduledforredaploy .

Reply:
Its not a singular state though. Redeploying and Upgrading will be different for Deployment objects since we need to unpause and then pause again for upgrades and scale down and up for redeploys. Maybe I can move it all into the UpgradePhase though? Will play around with that

Reply:
this will be further addressed by https://github.com/openshift/elasticsearch-operator/pull/92